src/node.cc

```diff
@@ -137,6 +141,8 @@ using v8::Value;
 
 namespace per_process {
 
+extern bool linux_at_secure;
```

Comment:
Moved this here since there seemed no reason for this to be specific to calls through `main()`.

Reply:
I think it's initialized there to make sure that no parsing of environment variables bypass the AT_SECURE check (provided that they use `SafeGetEnv()`. With this patch the `NODE_DEBUG_NATIVE` parsing would bypass it now, perhaps we should move the initialization of `linux_at_secure` to the beginning of `InitializeOncePerProcess()`, or add some comment to advise against adding code that retrieve the environment variables before PlatformInit() is called  - bypassing the check for `NODE_DEBUG_NATIVE` is probably OK anyway because it's entirely internal, but people might add add parsing of other non-internal variables there in the future.

Reply:
Yeah, thanks for bringing this up :slightly_smiling_face: I’ve decided to just be sure in 46057ce300e1738ba268b61fbf65b2f7f9019dff and initialize it on first access and make it independent of Node.js initialization.

Technically we could go one step further and just call `getauxval()` as part of static initialization, with the only downside being that it then is not usable in static initialization in other files. But since that is just hypothetical right now, lmk if you’d prefer that.

Reply:
The currently implementation is good enough IMO 👍